rclpy/src/rclpy/_rclpy.c

```diff
@@ -2766,7 +2744,7 @@ rclpy_wait(PyObject * Py_UNUSED(self), PyObject * args)
   if (!PyArg_ParseTuple(args, "O|K", &pywait_set, &timeout)) {
     return NULL;
   }
-  rcl_wait_set_t * wait_set = (rcl_wait_set_t *)PyCapsule_GetPointer(pywait_set, "rcl_wait_set_t");
+  rcl_wait_set_t * wait_set = PyCapsule_GetPointer(pywait_set, "rcl_wait_set_t");
```

Comment:
Good question. Currently, the `rcl_wait_set_t` capsule didn't have a deleter, and deletion was handled manually in `Python`.
To avoid doing this PR bigger, I didn't refactor how wait sets were used. It should be done in a follow up PR.

Reply:
Let's open an issue to track that work.